tests/test_duplicate_command.rs

```diff
@@ -89,6 +89,144 @@ fn test_duplicate() {
     "###);
 }
 
+#[test]
+fn test_duplicate_many() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "a", &[]);
+    create_commit(&test_env, &repo_path, "b", &["a"]);
+    create_commit(&test_env, &repo_path, "c", &["a"]);
+    create_commit(&test_env, &repo_path, "d", &["c"]);
+    create_commit(&test_env, &repo_path, "e", &["b", "d"]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @   921dde6e55c0   e
+    |\  
+    o | ebd06dba20ec   d
+    o | c0cb3a0b73e7   c
+    | o 1394f625cbbd   b
+    |/  
+    o 2443ea76b0b1   a
+    o 000000000000   (no description set)
+    "###);
+
+    // BUG! Copy of e shouldn't have b as a parent!
+    let stdout = test_env.jj_cmd_success(&repo_path, &["duplicate", "b:"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Duplicated 1394f625cbbd as 3b74d9691015 b
+    Duplicated 921dde6e55c0 as 8348ddcec733 e
+    "###);
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    o   8348ddcec733   e
+    |\  
+    o | 3b74d9691015   b
+    | | @   921dde6e55c0   e
+    | | |\  
+    | |/ /  
+    | o | ebd06dba20ec   d
+    | o | c0cb3a0b73e7   c
+    |/ /  
+    | o 1394f625cbbd   b
+    |/  
+    o 2443ea76b0b1   a
+    o 000000000000   (no description set)
+    "###);
+
+    test_env.jj_cmd_success(&repo_path, &["undo"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["duplicate", "b:", "d"]);
```

Comment:
We should also have a test of `jj duplicate b b` or `jj duplicate b: d:` or similar, i.e. where some revisions are mentioned multiple times.